asyncapi-cli/src/main/java/io/ballerina/asyncapi/cmd/AsyncApiCmd.java

```diff
@@ -124,6 +220,146 @@ public void execute() {
+        if (generatedFileType) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, JSON_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+    }
+
+    private void ballerinaToAsyncApiWs(String fileName) {
+        List<AsyncApiConverterDiagnostic> errors = new ArrayList<>();
+        final File balFile = new File(fileName);
+        Path balFilePath = null;
+        try {
+            balFilePath = Paths.get(balFile.getCanonicalPath());
+        } catch (IOException e) {
+            DiagnosticMessages message = DiagnosticMessages.AAS_CONVERTOR_102;
+            ExceptionDiagnostic error = new ExceptionDiagnostic(message.getCode(),
+                    message.getDescription(), null, e.getLocalizedMessage());
+            errors.add(error);
+        }
+        getTargetOutputPathWs();
+        // Check service name it is mandatory
+        BallerinaToAsyncApiGenerator asyncApiConverter = new BallerinaToAsyncApiGenerator();
+        asyncApiConverter.generateAsyncAPIDefinitionsAllService(balFilePath, targetOutputPath, service,
+                generatedFileType);
+
+        errors.addAll(asyncApiConverter.getErrors());
+        if (!errors.isEmpty()) {
+            for (AsyncApiConverterDiagnostic error : errors) {
+                if (error instanceof ExceptionDiagnostic exceptionDiagnostic) {
+                    AsyncAPIDiagnostic diagnostic = CmdUtils.constructAsyncAPIDiagnostic(exceptionDiagnostic.getCode(),
+                            exceptionDiagnostic.getMessage(), exceptionDiagnostic.getDiagnosticSeverity(),
+                            exceptionDiagnostic.getLocation().orElse(null));
+                    outStream.println(diagnostic);
+                    exitError(this.exitWhenFinish);
+                } else if (error instanceof IncompatibleRemoteDiagnostic incompatibleError) {
+                    AsyncAPIDiagnostic diagnostic = CmdUtils.constructAsyncAPIDiagnostic(incompatibleError.getCode(),
+                            incompatibleError.getMessage(), incompatibleError.getDiagnosticSeverity(),
+                            incompatibleError.getLocation().get());
+                    outStream.println(diagnostic);
+                }
+            }
+        }
+    }
+
+    private void asyncApiToBallerinaWs(String fileName) throws IOException {
+        AsyncApiToBallerinaGenerator generator = new AsyncApiToBallerinaGenerator();
+        generator.setLicenseHeader(this.setLicenseHeaderWs());
+        generator.setIncludeTestFiles(this.includeTestFiles);
+        final File asyncApiFile = new File(fileName);
+        getTargetOutputPathWs();
+        Path resourcePath = Paths.get(asyncApiFile.getCanonicalPath());
+        generatesClientFileWs(generator, resourcePath);
+    }
+
+    /**
+     * This util is to set the license header content which is to be added at the beginning of the ballerina files.
+     */
+    private String setLicenseHeaderWs() {
+        String licenseHeader = "";
```

Comment:
Any reason to initialize an empty String here. I think this method could have returned an empty Optional in case of an exception.

Reply:
Here, we expect the license header to be empty in most cases. Therefore we initially have the license header to be an empty string, and only replace it with the actual license header if it is provided.